common/lanelet2_extension/CMakeLists.txt

```diff
@@ -88,88 +115,122 @@ else() # Default to ROS 1 logging
   
 endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  ${deps_using_buildtool}
-)
 
-message("Logger includes: " ${${LOGGER_DEP}_INCLUDE_DIRS})
-message("Logger Libs: " ${${LOGGER_DEP}_LIBRARIES})
+if(${ROS_VERSION} EQUAL 2)
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES lanelet2_extension_lib
-  CATKIN_DEPENDS ${catkin_depends_deps}
-  DEPENDS Boost ${LOGGER_DEP}
-)
+  find_package(ament_cmake_auto REQUIRED)
 
-include_directories(
-  include
-  ${GeographicLib_INCLUDE_DIRS}
-  ${catkin_INCLUDE_DIRS}
-  ${PUGIXML_INCLUDE_DIRS}
-  ${PROJ4_INCLUDE_DIRS}
-  ${Boost_INCLUDE_DIRS}
-  ${${LOGGER_DEP}_INCLUDE_DIRS}
-)
+  ament_auto_find_build_dependencies()
 
-add_definitions (${GeographicLib_DEFINITIONS})
+  find_package(hardcoded_params REQUIRED)
 
-add_library( lanelet2_extension_lib
-  lib/autoware_osm_parser.cpp
-  lib/autoware_traffic_light.cpp
-  lib/local_frame_projector.cpp
-  lib/mgrs_projector.cpp
-  lib/CarmaUSTrafficRules.cpp
-  lib/CarmaTrafficLight.cpp
-  lib/PassingControlLine.cpp
-  lib/StopRule.cpp
-  lib/DigitalSpeedLimit.cpp
-  lib/DigitalMinimumGap.cpp
-  lib/RegionAccessRule.cpp
-  lib/DirectionOfTravel.cpp
-)
+  # Includes
+  include_directories(
+    include
+    ${GeographicLib_INCLUDE_DIRS}
+    ${PUGIXML_INCLUDE_DIRS}
+    ${PROJ4_INCLUDE_DIRS}
+    ${Boost_INCLUDE_DIRS}
+    ${${LOGGER_DEP}_INCLUDE_DIRS}
+    ${hardcoded_params_INCLUDE_DIRS}
+  )
 
-add_dependencies(lanelet2_extension_lib
-  ${${PROJECT_NAME}_EXPORTED_TARGETS}
-  ${catkin_EXPORTED_TARGETS}
-)
+  add_definitions (${GeographicLib_DEFINITIONS})
+
+  # Build
+  ament_auto_add_library(lanelet2_extension_lib SHARED
+    ${cpp_files}
+  )
+
+
+  target_link_libraries(lanelet2_extension_lib
+    ${GeographicLib_LIBRARIES}
+    ${PROJ4_LIBRARIES}
+    ${Boost_LIBRARIES}
+    ${${LOGGER_DEP}_LIBRARIES}
+  )
+
+
+  # TODO make unit tests compile with ament
```

Comment:
Is this something that can be done for this PR?

Reply:
I don't think so, its currently using gmock which wasn't available in ament yet as far as I know. Since the tests still exist in catkin they can be run until we remove ros1 from the system which is still a few releases away